File: Melo/Melo.Models/Models/PagedResponse.cs
namespace Melo.Models
{
	public class PagedResponse<TResponse>
	{
		public int Page {  get; set; }

		public int? PrevPage { get; set; }

		public int? NextPage { get; set; }

		public int TotalPages { get; set; }

		public int TotalItems { get; set; }

		public int Items {  get; set; }

		public List<TResponse> Data { get; set; } = new List<TResponse>();
	}
}

File: Melo/Melo.UI/melo_desktop/lib/models/paged_response.dart
class PagedResponse<T> {
  final int page;
  final int? prevPage;
  final int? nextPage;
  final int totalPages;
  final int totalItems;
  final int items;
  final List<T> data;

  PagedResponse({
    required this.page,
    this.prevPage,
    this.nextPage,
    required this.totalPages,
    required this.totalItems,
    required this.items,
    required this.data,
  });

  factory PagedResponse.fromJson(
    Map<String, dynamic> json,
    T Function(Map<String, dynamic>) fromJsonT,
  ) {
    return PagedResponse(
      page: json['page'],
      prevPage: json['prevPage'],
      nextPage: json['nextPage'],
      totalPages: json['totalPages'],
      totalItems: json['totalItems'],
      items: json['items'],
      data: (json['data'] as List).map((e) => fromJsonT(e)).toList(),
    );
  }
}
